cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -220,15 +221,19 @@ public QueryHandle enableDataset(Id.DatasetInstance datasetID, DatasetSpecificat
   }
 
   private QueryHandle createFromSchemaProperty(DatasetSpecification spec, Id.DatasetInstance datasetID,
-                                               Map<String, String> serdeProperties)
+                                               Map<String, String> serdeProperties, boolean shouldErrorOnMissingSchema)
     throws ExploreException, SQLException, UnsupportedTypeException {
 
     String schemaStr = spec.getProperty(DatasetProperties.SCHEMA);
     // if there is no schema property, we cannot create the table and this is an error
     if (schemaStr == null) {
-      throw new IllegalArgumentException(String.format(
-        "Unable to enable exploration on dataset %s because the %s property is not set.",
-        datasetID.getId(), DatasetProperties.SCHEMA));
+      if (shouldErrorOnMissingSchema) {
```

Comment:
preconditions should be used to check conditions that should always be true.  this is more like bad user input. It should probably actually be throwing BadRequestException, but leaving that exception refactoring to another PR.